base/Base.jl

```diff
@@ -6,10 +6,19 @@ using Core.Intrinsics, Core.IR
 
 # to start, we're going to use a very simple definition of `include`
 # that doesn't require any function (except what we can get from the `Core` top-module)
-const _included_files = Array{Tuple{Module,String},1}(Core.undef, 1)
+# start this big so that we don't have to resize before we have defined how to grow an array
+const _included_files = Array{Tuple{Module,String},1}(Core.undef, 50)
+_included_files.size = (1,)
```

Comment:
Just a small thing — perhaps we should specialize `getproperty` to make `Array` look fully opaque from the REPL (and require get/set field for this kind of internals muckery).  Given other languages, `arr.size` is a really tempting footgun.

Reply:
yeah that's a good point. As nice as it made coding these, they should probably require a `setfield`

Reply:
Perhaps a name other than `size` could serve an additional purpose, to further reduce "just access this"? Something like `dims_tuple`?

Reply:
What does `setfield` has to do with REPL autocompletions? 

And what's wrong with Matt's original suggestion again?

Reply:
what I mean is that if we overload `getproperty` (and presumably `setproperty`) you would need `setfield` to modify the size. I was agreeing with the original suggestion.

Reply:
Isn't it enough to overload `propertynames` which the REPL use for autocomplete?